### PR TITLE
DISA STIG compliance

### PR DESCRIPTION
**Pull request by fwilhe**
**What this PR does / why we need it**:

Add a feature to make the Garden Linux image as compliant as possible to the [Canonical Ubuntu 20.04 LTS Security Technical Implementation Guide](https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/)

Note that some items in that STIG simply don't apply to Garden Linux, for example it does not contain a GUI or AppArmor.

STIG-specific configuration is encapsulated in the 'stig' feature. To opt in into it, build your image with that feature enabled. Note that this might break other features.

The 'stigDev' feature provides a known user/password and ssh key for the purpose of testing. This feature is only intended for development/testing purposes, not for production.

### CODE REVIEW

**Review by Vincinator — COMMENTED** [review]
Added my feedback. 

Let's have a chat if you need to get this in. We can collaborate on the builder PR, and/or discuss whether we need it. 

Despite the nit-picks and general questions: very clean PR

File: features/stig/file.include/etc/pam.d/common-auth [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/finding/V-238237
+# Disabled because it breaks the _dev feature
+# auth required pam_faildelay.so delay=4000000
```
[the remark this answers is not in the document]
Comment (fwilhe) [reply features/stig/file.include/etc/pam.d/common-auth]: No it's actually taken from https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/finding/V-238237

But putting the files into /etc/pam.d turned out to be wrong, need to rework that.

File: features/stig/exec.config [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+set -x
+
+ensure_line() {
+    local file="${1}"; shift
+    local property="${1}"; shift
+    local parameters="${1}"; shift
+
+    if [[ -f "$file" ]]; then
+        if grep -q "^$property" "$file"; then
+            sed -i "s/^\($property \).*/\1$parameters/" "$file"
```
[the remark this answers is not in the document]
Comment (fwilhe) [reply features/stig/exec.config:13]: thanks for verifying

File: features/stig/exec.config [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    # https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/finding/V-255912
+    ensure_line /etc/ssh/sshd_config KexAlgorithms ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256
+
+    # https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/finding/V-238380
+    #systemctl disable ctrl-alt-del.target
+    #systemctl mask ctrl-alt-del.target
+    #systemctl daemon-reload
+
+    # https://www.stigviewer.com/stig/canonical_ubuntu_20.04_lts/2023-09-08/finding/V-238208
+    ensure_file_absent /etc/sudoers.d/wheel
```
[the remark this answers is not in the document]
Comment (fwilhe) [reply features/stig/exec.config:62]: I'm working on the `stigDev` feature which is purpose-built for making tests possible. This is WIP.

**Review by Vincinator — APPROVED** [review]
Looks good to me merged as independent feature

**Pull request merged**